sim-cli/src/main.rs

```diff
@@ -199,3 +217,47 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+async fn read_sim_path(data_dir: PathBuf, sim_file: PathBuf) -> anyhow::Result<PathBuf> {
+    let sim_path = if sim_file.is_relative() {
+        data_dir.join(sim_file)
+    } else {
+        sim_file
+    };
+
+    if sim_path.exists() {
+        Ok(sim_path)
+    } else {
+        log::info!("Simulation file '{}' does not exist", sim_path.display());
+        let sim_files: Vec<String> = std::fs::read_dir(data_dir.clone())?
+            .filter_map(|f| {
+                f.ok().and_then(|f| {
+                    if f.path().extension()?.to_str()? == "json" {
+                        return f.file_name().into_string().ok();
+                    }
+                    None
+                })
+            })
+            .collect::<Vec<_>>();
+
+        if sim_files.is_empty() {
+            anyhow::bail!("no simulation files found in {}.", data_dir.display());
+        }
+
+        let selection = dialoguer::Select::new()
+            .with_prompt(format!(
+                "Select a simulation file. Found these in {}:",
```

Comment:
This also reads pretty confusing:

```
Select a simulation file. Found these in .::
```

The trailing `:` needs to be removed, given `dialoguer` already adds it, furthermore, ending the prompt with a `.` would still make it look weird. In this case I'd get rid os the format altogether or expand it to absolute